src/scheduler/fifo.c

```diff
@@ -1242,17 +1242,12 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 					snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
 					schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
 				}
-				if (throughput)
-					rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-				else
-					rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 			}
-		} else {
-			if (throughput)
-				rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-			else
-				rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 		}
+		if (rc > 0 && throughput)
+			rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
+		else
```

Comment:
This looks wrong! When rc turns out to be negative we will end up calling pbs_runjob().
Either put whole if/else related to throughput inside another condition that checks for rc to be greater than 0 OR go back to original implementation.